workflow/AlertDB.go

```diff
@@ -117,27 +117,23 @@ func (db *AlertDB) GetAllByTrigger(format *piazza.JsonPagination, triggerId stri
 	return alerts, searchResult.TotalHits(), nil
 }
 
-func (db *AlertDB) GetOne(id piazza.Ident) (*Alert, error) {
+func (db *AlertDB) GetOne(id piazza.Ident) (*Alert, bool, error) {
 	getResult, err := db.Esi.GetByID(db.mapping, id.String())
 	if err != nil {
-		return nil, LoggedError("AlertDB.GetOne failed: %s", err)
+		return nil, getResult.Found, LoggedError("AlertDB.GetOne failed: %s", err)
 	}
 	if getResult == nil {
-		return nil, LoggedError("AlertDB.GetOne failed: no getResult")
-	}
-
-	if !getResult.Found {
-		return nil, nil
+		return nil, true, LoggedError("AlertDB.GetOne failed: no getResult")
```

Comment:
the bool is `found`. This is the switch that determines whether or not an error is a 404 or a 400. If !Found, there will always be an error with the implementation of GetByID. I think it is actually impossible for get result to be nil, just kept it there to be on the safe side